### PR TITLE
DEV-3143: Remove Anymail

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Removes the django-anymail package and uses Django's built-in SMTP handling instead.

#### Why are we doing this? How does it help us?

Reduces complexity, gives us flexibility to change email providers more easily.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Updated a few tests that used Anymail-specific mocks.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3143

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

⚠️ https://news-revenue-hub.atlassian.net/browse/ITS-2688 needs to be done before this merges.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

There is also https://news-revenue-hub.atlassian.net/browse/ITS-2689 for cleanup after this merges and we're confident it's working.

### CODE REVIEW

File: revengine/settings/deploy.py [hunk 1]
```diff
@@ -16,15 +16,13 @@
...
-EMAIL_BACKEND = "anymail.backends.mailgun.EmailBackend"
-MAILGUN_SENDER_DOMAIN = "fundjournalism.org"
-
-ANYMAIL = {
-    "MAILGUN_API_KEY": os.getenv("MAILGUN_API_KEY", ""),
-    "MAILGUN_SENDER_DOMAIN": MAILGUN_SENDER_DOMAIN,
-}
-
+EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+EMAIL_HOST = os.getenv("EMAIL_HOST")
```
Comment (njharman) [inline revengine/settings/deploy.py:20]: Consider raising exception or critical log if required env vars are not (present and non-empty). os.getenv returns None on default and we'd want to look for empty string.

Those listed in [ITS-2688]

**Review by njharman — APPROVED** [review]

File: revengine/settings/deploy.py (hunk 1 above)
Comment (nrh-cklimas) [reply revengine/settings/deploy.py:20]: Leaving this here for future reviewers to weigh in on. My feeling is that if we have email misconfigured, we're going to find out pretty fast regardless.

**Review by benjaminEwhite — APPROVED** [review]

File: revengine/settings/deploy.py (hunk 1 above)
Comment (x110dc) [reply revengine/settings/deploy.py:20]: I do think it's a good practice to fail quickly if expected environment variables don't exist, which is one of the reasons we use `environs` on the Switchboard side (as that's the default with that package). But I'm okay deferring this to a separate ticket where we implement that package for RevEngine (or equivalent changes). If there's not already a ticket for that can you open one and add a code comment about it here @nrh-cklimas ?

**Review by x110dc — APPROVED** [review]
I'd also add to the "why are we doing this" section that this change makes it easier for us to change email providers. And that was the main point of this ticket.

**Pull request merged**